src/lib/Hydra/Plugin/GithubStatus.pm

```diff
@@ -32,23 +32,30 @@ sub common {
         my $ua = LWP::UserAgent->new();
 
         foreach my $conf (@config) {
+            print STDERR "GithubStatus_Debug job name $jobName\n";
             next unless $jobName =~ /^$conf->{jobs}$/;
             # Don't send out "pending" status updates if the build is already finished
             next if !$finished && $b->finished == 1;
 
             my $contextTrailer = $conf->{excludeBuildFromContext} ? "" : (":" . $b->id);
+            my $github_job_name = $jobName =~ s/-pr-\d+//r;
+            my $extendedContext = $conf->{context} // "continuous-integration/hydra:" . $jobName . $contextTrailer;
+            my $shortContext = $conf->{context} // "ci/hydra:" . $github_job_name . $contextTrailer;
+            my $context = $conf->{useShortContext} ? $shortContext : $extendedContext;
             my $body = encode_json(
                 {
                     state => $finished ? toGithubState($b->buildstatus) : "pending",
                     target_url => "$baseurl/build/" . $b->id,
                     description => $conf->{description} // "Hydra build #" . $b->id . " of $jobName",
-                    context => $conf->{context} // "continuous-integration/hydra:" . $jobName . $contextTrailer
+                    context => $context
                 });
             my $inputs_cfg = $conf->{inputs};
             my @inputs = defined $inputs_cfg ? ref $inputs_cfg eq "ARRAY" ? @$inputs_cfg : ($inputs_cfg) : ();
             my %seen = map { $_ => {} } @inputs;
             while (my $eval = $evals->next) {
+                print STDERR "GithubStatus_Debug eval $eval\n";
                 foreach my $input (@inputs) {
+                    print STDERR "GithubStatus_Debug input $input\n";
                     my $i = $eval->jobsetevalinputs->find({ name => $input, altnr => 0 });
```

Comment:
It isn't clear from these messages what they're trying to communicate. Could they be a bit more ... wordy? :)